_posts/2020-10-19-solidity-0.7.4-release-announcement.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.7.4 Release Announcement'
+date: '2020-10-19'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.7.4](https://github.com/ethereum/solidity/releases/tag/v0.7.4) fixes a storage corruption bug of medium severity.
+To learn more about the bug and to check if your contract is vulnerable please read this [post](https://solidity.ethereum.org/2020/10/19/empty-byte-array-copy-bug) with
+further details about the bug. The bug has been reported by John Toman of the Certora development team.
+Their bug disclosure post can be found [here](https://www.certora.com/blog/corruptedStorage.html).
+
+Additionally, v0.7.4 adds ....
+
+## Important Bugfixes
+ *  Code Generator: Fix data corruption bug when copying empty byte arrays from memory or calldata to storage. Read more [here](https://solidity.ethereum.org/2020/10/19/empty-byte-array-copy-bug).
+
+
+## Notable New Features
+
+### Constants at File-Level
+
+In order to make functions at file-level more useful, Solidity now also allows to
+define constant variables at file-level:
+
+```solidity
+uint constant EXPONENT = 10;
+uint constant MULTIPLIER = 2**EXPONENT;
+
+struct Fixed { uint value; }
+
+function toFixed(uint x) pure returns (Fixed memory ret) {
+    ret.value = x * MULTIPLIER;
+}
+```
+
+Such constants have to follow the same rules as constants at contract-level.
+This means, for example, the value has to be fixed at compile-time and only
+some types are supported.
+
+You can import constants from other source files like you import any other identifier:
+
+```import {MyConstant, MyStruct} from "myHelper.sol";```
+
+
+
+## Full Changelog
+
+### Language Features: 
```

Comment:
There were some more changes - please take this version: https://github.com/ethereum/solidity/pull/10065